index.js

```diff
@@ -151,11 +150,11 @@ module.exports = {
                     if (session) {
                         var filename = 'ScreenShot_onException-' + process.pid + '-' + new Date().getTime();
                         var screenShotFileName = path.resolve(screenShotPath, filename);
-                        flow.wait(function () {
-                            return nemo.screenshot.snap(screenShotFileName).then(function () {
-                                exception._nemoScreenshotHandled = true;
-                                throw exception;
-                            }, 10000);
+                        nemo.screenshot.snap(screenShotFileName).then(function () {
```

Comment:
After summarizing my test results, I would answer the original question by saying the flow.wait() command was never resulting in the desired behavior of taking the screenshot, throwing the underlying exception, and having mocha report that exception.